spec/models/admin_flash_spec.rb

```diff
@@ -1,75 +1,37 @@
 require 'rails_helper'
 
 describe AdminFlash do
-  let(:admin_flash) { AdminFlash.new(:message => "Lorem ipstum", :expires => 2.days.from_now) }
-  let(:admin_flash_2) { AdminFlash.new(:message => "Lorem ipstum_2", :expires => 3.days.from_now) }
-  let(:expired_admin_flash) { AdminFlash.create(:message => "Lorem ipstum", :expires => 2.days.ago) }
-
-  subject { admin_flash }
-
-  it { is_expected.to respond_to(:message) }
-  it { is_expected.to respond_to(:expires) }
-
-  it { is_expected.to be_valid }
-
-  describe "when 'message' is blank" do
-    before { admin_flash.message = " " }
-
-    it { is_expected.not_to be_valid }
+  let(:admin_flashes) { [admin_flash, expired_admin_flash] }
+  let(:admin_flash) {
+    double('AdminFlash', id: 1, message: 'foo', expires: 1.days.from_now)
+  }
+  let(:expired_admin_flash) {
+    double('AdminFlash', id: 1, message: 'bar', expires: 1.days.ago)
+  }
+  let(:current_time) { '2016-10-21 12:52' }
+
+  it { is_expected.to validate_presence_of(:message) }
+  it { is_expected.to validate_presence_of(:expires) }
+
+  before do
+    allow(AdminFlash).to receive(:order).with('created_at desc').
+      and_return(admin_flashes)
+
+    allow(Time).to receive(:now).and_return(current_time)
+
+    allow(admin_flashes).to receive(:where).with('expires >= ?', current_time).
+      and_return([admin_flash])
   end
 
   describe "#unexpired_messages" do
```

Comment:
I think it should read `describe ".unexpired_messages" do` as we're describing a class method.